### PR TITLE
update utility manage reports

### PR DESCRIPTION
**Pull request by Deveshs23**

### CODE REVIEW

File: cs.py [hunk 1]
```diff
@@ -69,7 +73,7 @@ def main():
             awsaudit.aws_audit()
             merger.merge()
             log.info("completed aws audit")
-            exit(0)
+        #    exit(0)
```
Comment (sandy724) [inline cs.py]: Removed this commented out code

File: cs.py [hunk 2]
```diff
@@ -82,8 +86,15 @@ def main():
         from modules import azureaudit
         azureaudit.azure_audit()
         log.info("completed azure audit")
+        #exit(0)
```
Comment (sandy724) [inline cs.py]: Again remove this commented out code

File: cs.py [hunk 3]
```diff
@@ -82,8 +86,15 @@ def main():
...
+
+    
+    if args.number:
```
Comment (sandy724) [inline cs.py]: What if a person gives both wide & number

File: modules/manageReports.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import pathlib
+import glob
+import time
+import shutil
+import logging
+
+def listOfFolder(number, directory):
+    try:
+        listOfReports = []
+        for i in range(number):
```
Comment (sandy724) [inline modules/manageReports.py]: As already said avoid using variables like i

File: modules/manageReports.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        return listOfReports
+    except Exception as identifier:
+        logging.exception(identifier)
+    
+
+def inverseList(directory, number):
+    try:
+        totalFolderList = sorted(glob.glob(os.path.join(directory, '*/')))
+        selectedFolder = listOfFolder(number, directory)
+        ListOfDeletingFolder = set(totalFolderList).difference(selectedFolder)
```
Comment (sandy724) [inline modules/manageReports.py]: follow proper variable naming convention ListOfDeletingFolder should be listOfDeletingFolder

File: modules/manageReports.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        aws(currentPath, number)
+        
+    elif(enviroment == 'azure'):
+        azure(currentPath, number)
+
+    elif(enviroment == 'gcp'):
+        gcp(currentPath, number)
+
+
+def aws(currentPath, number):
```
Comment (sandy724) [inline modules/manageReports.py]: cleanupReports(currentPath, number, cloudBasePath, accountId)
cleanupReports((currentPath, number, 'reports/AWS/aws_audit', awsaudit.account_name)
cleanupReports((currentPath, number, 'reports/GCP', awsaudit.account_name, gcpaudit.project_name)

File: modules/manageReports.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        selectedFolder = listOfFolder(number, directory)
+        ListOfDeletingFolder = set(totalFolderList).difference(selectedFolder)
+        return ListOfDeletingFolder
+    except Exception as identifier:
+        logging.exception(identifier)
+
+
+def deleteListOfFolder(directory, number):
+    
+    if(number < 0):
```
Comment (sandy724) [inline modules/manageReports.py]: Is this a correct place?

File: cs.py [hunk 8]
```diff
@@ -82,8 +86,15 @@ def main():
...
+        
+        from modules import manageReports
```
Comment (sandy724) [inline cs.py]: Why we are importing module here

File: modules/manageReports.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+            for i in deletingList:
+                shutil.rmtree(i)
+        except IndexError:
+            pass
+
+
+def numberOfReports(enviroment, number):
+    currentPath = os.getcwd()
+    
+    if(enviroment == 'aws'):
```
Comment (sandy724) [inline modules/manageReports.py]: Prefer case over if

File: cs.py (hunk 8 above)
Comment (iamabhishek-dubey) [reply cs.py]: Yes import module should go in top

File: modules/manageReports.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+import os
+import pathlib
+import glob
+import time
+import shutil
+import logging
+
+def listOfFolder(number, directory):
+    try:
```
Comment (iamabhishek-dubey) [inline modules/manageReports.py]: Change function name to snake-case as per python standard. Like:-
```python
def list_folders():
```

File: modules/merger.py [hunk 11]
```diff
@@ -314,4 +314,4 @@ def merge():
     webbrowser.open('file://' + os.path.realpath("./reports/AWS/aws_audit/%s/%s/final_report/report.html")
                     % (account_name, timestmp))
     fin = os.path.realpath("./reports/AWS/aws_audit/%s/%s/final_report/report.html") % (account_name, timestmp)
-    print ("THE FINAL REPORT IS LOCATED AT -------->  %s" % (fin))
+    print ("THE FINAL REPORT IS LOCATED AT -------->  %s" % (fin))
```
Comment (iamabhishek-dubey) [inline modules/merger.py:317]: Add a blank line at EOF

File: cs.py [hunk 12]
```diff
@@ -82,8 +86,15 @@ def main():
...
+        manageReports.numberOfReports(args.environment, args.number)
         exit(0)
```
Comment (iamabhishek-dubey) [inline cs.py:108]: Extra lines not required. Can you please use `pylint` to check your code quality. I think we can make some improvements

**Review by iamabhishek-dubey — CHANGES_REQUESTED** [review]

**Pull request closed without merging**